apps/server/src/shared/domain/entity/task-card.entity.ts

```diff
@@ -67,6 +75,36 @@ export class TaskCard extends BaseEntityWithTimestamps implements ICard, ITaskCa
 		return this.visibleAtDate < this.dueDate;
 	}
 
-	@OneToOne({ type: 'Task', fieldName: 'taskId', eager: true, unique: true, cascade: [Cascade.ALL] })
-	task!: Task;
+	public completeForUser(user: User): void {
+		this.completed.add(user);
+	}
+
+	public undoForUser(user: User): void {
```

Comment:
rename to removeUserFromCompletedList

also both add & remove functions seems to be unnecessary. The caller should just operate directly on the collection. 
But I have no strong opinion on this.